exporter/collector/logs.go

```diff
@@ -214,6 +224,7 @@ func (l *LogsExporter) PushLogs(ctx context.Context, ld plog.Logs) error {
 
 			entries = entries[entry:]
 			entry = 0
+			currentBatchSize = 0
```

Comment:
confirmed this was the fix by commenting this line out which reproduced the infinite loop on the new test case